template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Areas/IdentityService/Extensions/IdentityServiceExtensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Identity.Service.Extensions
+{
+    public static class IdentityServiceExtensions
+    {
+        public static IIdentityServiceBuilder AddIdentityService<TUser, TApplication>(this IServiceCollection services)
+            where TUser : class
+            where TApplication : class
+        {
+            services.AddSingleton<IConfigureOptions<IdentityServiceOptions>, IdentityServiceOptionsConfigurationSetup>();
+            return services.AddIdentityService<TUser, TApplication>(_ => { });
+        }
+
+        public static IIdentityServiceBuilder AddIdentityServiceExtensions(this IIdentityServiceBuilder builder)
+        {
+            builder.Services.AddScoped<IAuthorizationResponseParameterProvider, ClientInfoProvider>();
+            builder.Services.AddScoped<ITokenResponseParameterProvider, ClientInfoProvider>();
+            builder.Services.Configure<IdentityServiceOptions>(options =>
+            {
+                AddContextClaims(options.IdTokenOptions.ContextClaims);
+                AddContextClaims(options.AccessTokenOptions.ContextClaims);
+            });
+            return builder;
+        }
+
+        private static void AddContextClaims(TokenMapping tokenMapping)
+        {
+            tokenMapping.AddSingle(IdentityServiceExtensionsClaimTypes.TrustFrameworkPolicy, IdentityServiceExtensionsAmbientClaimTypes.Policy);
+            tokenMapping.AddSingle(IdentityServiceExtensionsClaimTypes.Version, IdentityServiceExtensionsAmbientClaimTypes.Version);
+        }
+
+        public static void AddExtensionsAmbientClaims(
+            this TokenGeneratingContext context, string policy, string version, string tenantId)
+        {
+            context.AmbientClaims.Add(new Claim(IdentityServiceExtensionsAmbientClaimTypes.Policy, policy));
```

Comment:
I don't think so. These are interim claims that will get mapped to the real claims that get put on the wire.